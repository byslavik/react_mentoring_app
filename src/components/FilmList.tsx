import * as React from 'react';

import {FilmCard} from './FilmCard';
import {PersonCard} from './PersonCard';
import {NoItems} from './NoItems';
import * as s from './scss/FilmList.scss';

import {Film} from '../models/film.model';
import {Person} from '../models/person.model';
import {searchMethod} from '../models/common.models';

interface FilmListProps {
  films: Film[] | Person[];
  method: searchMethod;
}


export class FilmList extends React.Component<FilmListProps, any> {

  buildCards(data: Film[] | Person[])  {
    const CardComponent = this.props.method == 'movie' ? FilmCard : PersonCard;
    
    return data.map((item: Film | Person, index) => {
      return <CardComponent key={item.id} item={item} />;
    })
  }

  render() {
    const cards:JSX.Element[] = this.buildCards(this.props.films);
    return <div className={s.filmList}>
      {cards.length != 0 ? cards : <NoItems/> }
    </div> 
  }
}





