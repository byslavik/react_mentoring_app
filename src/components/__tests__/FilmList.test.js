import * as React from 'react';
import {FilmList} from '../FilmList';
import * as renderer from 'react-test-renderer';
import {filmList} from './testData'

jest.mock('react-router-dom',()=> ({Link:'Link'}));
jest.mock('../FilmCard',()=> ({FilmCard: 'FilmCard'}));
jest.mock('../PersonCard',()=> ({PersonCard: 'PersonCard'}));
jest.mock('../NoItems',()=> ({NoItems: 'NoItems'}));

describe('FilmList component', ()=>{
  it('FilmList component shows correctly with FilmData', () => {
    const tree = renderer.create(
      <FilmList films={filmList} method="movie"/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('FilmList component shows correctly with PersonData', () => {
    const tree = renderer.create(
      <FilmList films={filmList} method="person"/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('FilmList component shows correctly without data', () => {
    const tree = renderer.create(
      <FilmList films={[]} method="person"/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});