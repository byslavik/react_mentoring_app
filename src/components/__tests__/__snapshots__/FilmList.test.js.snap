// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmList component FilmList component shows correctly with FilmData 1`] = `
<div
  className={undefined}
>
  <FilmCard
    item={
      Object {
        "adult": false,
        "backdrop_path": "/somepath.jpg",
        "id": 1,
        "name": "Name",
        "original_title": "Original Title",
        "overview": "Some description",
        "popularity": 100500,
        "poster_path": "/somepath.jpg",
        "profile_path": "/somepath.jpg",
        "title": "Title",
        "video": false,
        "vote_average": 5,
        "vote_count": 1000,
      }
    }
  />
</div>
`;

exports[`FilmList component FilmList component shows correctly with PersonData 1`] = `
<div
  className={undefined}
>
  <PersonCard
    item={
      Object {
        "adult": false,
        "backdrop_path": "/somepath.jpg",
        "id": 1,
        "name": "Name",
        "original_title": "Original Title",
        "overview": "Some description",
        "popularity": 100500,
        "poster_path": "/somepath.jpg",
        "profile_path": "/somepath.jpg",
        "title": "Title",
        "video": false,
        "vote_average": 5,
        "vote_count": 1000,
      }
    }
  />
</div>
`;

exports[`FilmList component FilmList component shows correctly without data 1`] = `
<div
  className={undefined}
>
  <NoItems />
</div>
`;
